lntest/itest/lnd_single_hop_invoice_test.go

```diff
@@ -161,6 +162,68 @@ func testSingleHopInvoice(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf(err.Error())
 	}
 
+	// Now create an invoice and specify routing hints.
+	// We will test that the routing hints are encoded properly.
+	hintChannel := lnwire.ShortChannelID{BlockHeight: 10}
+	bobPubKey := hex.EncodeToString(net.Bob.PubKey[:])
+	hints := []*lnrpc.RouteHint{
+		{
+			HopHints: []*lnrpc.HopHint{
+				{
+					NodeId:                    bobPubKey,
+					ChanId:                    hintChannel.ToUint64(),
+					FeeBaseMsat:               1,
+					FeeProportionalMillionths: 1000000,
+					CltvExpiryDelta:           20,
+				},
+			},
+		},
+	}
+
+	invoice = &lnrpc.Invoice{
+		Memo:       "hints",
+		Value:      paymentAmt,
+		RouteHints: hints,
+	}
+
+	ctxt, _ = context.WithTimeout(ctxt, defaultTimeout)
+	invoiceResp, err = net.Bob.AddInvoice(ctxt, invoice)
+	if err != nil {
+		t.Fatalf("unable to add invoice: %v", err)
+	}
+	payreq, err := net.Bob.DecodePayReq(ctxt, &lnrpc.PayReqString{PayReq: invoiceResp.PaymentRequest})
+	if err != nil {
+		t.Fatalf("failed to decode payment request %v", err)
+	}
+	if len(payreq.RouteHints) != 1 {
+		t.Fatalf("expected one routing hint")
+	}
+	routingHint := payreq.RouteHints[0]
```

Comment:
could we use require.Equal(payreq.RouteHints, hints)?

Reply:
@carlaKC  would love to but comparing protobuf generated structures by value in't really working since they contain "hidden" generated fields which are assigned with different values for different instances (such as `XXX_sizecache`)

Reply:
ah sneaky, no worries then!